docs/advanced.rst

```diff
@@ -176,6 +176,57 @@ are listed below:
 +-----------+--------------------------+---------------------------+
 
 
+Handling Board Variants With a Single Bundle
+--------------------------------------------
+
+If you have variants of hardware that require to install different variants of
+one image (e.g. for the kernel or for and FPGA bitstream) but have other images 
+that are common (such as the rootfs) between all hardware variants,
+RAUC allows you to put multiple different variants of these image into the same
+bundle at a time.
+RAUC calls this feature 'image variants'.
+
+.. image:: images/rauc-image-variants.svg
+  :width: 300
+
+If you want to make use of image variants, you first of all need to say which
+variant your specific board is. You can do this in your ``system.conf`` by
+setting the ``variant`` key:
+
+.. code-block:: cfg
+
+  [system]
+  ...
+  variant=variant-1
+
+The ``variant`` key accepts a special value ``dtb`` that allows (on device-tree
+based boards) to use the systems compatible string as board variant.
+
+.. code-block:: cfg
+
+  [system]
+  ...
+  variant=dtb
+
+In your manifest, you can specify variants of an image (e.g. the kernel here) as
+follows:
+
+.. code-block:: cfg
+
+  [image.kernel.variant-1]
+  filename=variant1.img
+  ...
+
+  [image.kernel.variant-2]
+  filename=variant1.img
+  ...
+
+It is allowed to have both a specific variant as well as a default image in the
+same bundle.
+Then, on a system that matches a specific variant an image is available for
```

Comment:
> If a specific variant of the image is available, it will be used on that system.
> ￼On all other systems, the default image will be used instead.

Also document that if an image exists in the manifest, it should be available for all variants. Either using a default or by listing all variants.